tests/hwtracer_ykpt/void_ret.c

```diff
@@ -0,0 +1,21 @@
+// Run-time:
+
+#include <stdio.h>
+#include <yk_testing.h>
+
+void __attribute__((noinline)) f() {
+  fputs("inside f\n", stderr);
+  return;
+}
+
+int main(void) {
+  int i = 4;
+  NOOPT_VAL(i);
+
+  void *tc = __hwykpt_start_collector();
+  while (i-- > 0)
```

Comment:
Isn't this a `for` loop in disguise?

Reply:
Kind of.

It can't be the for loop we all want it to be:
```
/home/vext01/research/yk/tests/hwtracer_ykpt/void_ret.c:13:16: error: expected expression
  for (int i = NOOPT_VAL(4); i > 0; i--)
               ^
```

`NOOPT_VAL` isn't an expression (it's inline asm), so it hoses that idea.

So, it would have to be this cursed loop instead:
```
int i = 4;
NOOPT_VAL(4);
...
for (; i > 0; i--) {
    ...
```

At which point I find what we started with more concise and clearer:

```
int i = 4;
NOOPT_VAL(4);
...
while (i-- > 0)
    ...
```